src/System.Windows.Forms.Primitives/src/System/Windows/Forms/GdiPlus/GdiPlusCache.cs

```diff
@@ -0,0 +1,140 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Drawing;
+
+namespace System.Windows.Forms
+{
+    /// <summary>
+    ///  Cache of GDI+ objects to reuse commonly created items.
+    /// </summary>
+    internal static class GdiPlusCache
+    {
+        // Keeping these as ThreadStatic for now. With testing, and if we can make them immutable ourselves (needs
+        // additional System.Drawing functionality), we can look at making this shared. Adding locking, however, does
+        // add cost to retrieving entries so it might not be worth it.
+        [ThreadStatic]
+        private static PenCache? s_penCache;
+        [ThreadStatic]
+        private static SolidBrushCache? s_brushCache;
+
+        // Picking soft and hard limits expecting that 30 of each is a reasonable upper limit of permanently reserved
+        // space. These items are roughly .5KB each, with _most_ of the memory being native (in the GDI+ object).
+
+        private static PenCache PenCache => s_penCache ??= new PenCache(softLimit: 30, hardLimit: 40);
+        private static SolidBrushCache BrushCache => s_brushCache ??= new SolidBrushCache(softLimit: 30, hardLimit: 40);
+
+        private static PenCache.Scope GetPen(Color color)
```

Comment:
This bothers me, for API readability and clarity of the intent should we add `Scope` suffix?
```suggestion
        private static PenCache.Scope GetPenScope(Color color)
```

Reply:
Sure